src/components/domain/survey/SurveyPage.tsx

```diff
@@ -0,0 +1,202 @@
+import { useState, useEffect } from 'react';
+import { useForm, FormProvider, SubmitHandler } from 'react-hook-form';
+import { useSubmitSurvey } from '@/hooks/queries/useSurveyService';
+import {
+  SurveyStep,
+  SURVEY_QUESTION_TYPE,
+  SurveyQuestionCategoryType,
+} from '@/models/survey/surveyModels';
+import {
+  SurveyLinkResponse,
+  SubmitSurveyRequest,
+  SurveyFormValues,
+} from '@/models/survey/surveyApiModels';
+import { surveyQuestions } from '@/lib/surveyQuestions';
+import SurveyIntroduction from '@/components/domain/survey/SurveyIntroduction';
+import RatingAnswer from '@/components/domain/survey/answerType/RatingAnswer';
+import CheckBoxAnswer from '@/components/domain/survey/answerType/CheckBoxAnswer';
+import TextAnswer from '@/components/domain/survey/answerType/TextAnswer';
+import MessageBox from '@/components/common/messgeBox/MessageBox';
+import {
+  Carousel,
+  CarouselContent,
+  CarouselItem,
+  CarouselNext,
+  CarouselPrevious,
+  type CarouselApi,
+} from '@/components/ui/carousel';
+import { Button } from '@/components/ui/button';
+import { MailOpen } from 'lucide-react';
+
+interface SurveyPageProps {
+  surveyData: SurveyLinkResponse;
+}
+
+export default function SurveyPage({ surveyData }: SurveyPageProps) {
+  const [current, setCurrent] = useState<number>(1);
```

Comment:
```suggestion
  const [currentStep, setCurrentStep] = useState<number>(1);
```
명확하게 해주는게 덜 헷갈릴 것 같네요! 근데 이 부부은 자잘한거라 수정 안해주셔도 됩니당